nmosregistration/aggregation.py

```diff
@@ -38,23 +39,33 @@
 
 class AggregatorAPI(WebAPI):
 
-    def __init__(self, logger, config, registry=EtcdInterface()):
+    def __init__(self, logger, config):
+        if config['registry']['type'] == 'couchbase':
+            registry = CouchbaseInterface(
+                cluster_address=config['registry']['hosts'],
+                username=config['registry']['username'],
+                password=config['registry']['password'],
+                bucket=config['registry']['bucket']
+            )
+        else:
+            registry = EtcdInterface()
         super(AggregatorAPI, self).__init__()
         self._config = config
 
-        garbage_collect_interval = int(self._config.get("garbage_collect_interval", 10))
-        self._garbage_collector = GarbageCollect(identifier=HOST, registry=registry, interval=garbage_collect_interval)
+        if config['registry']['type'] == 'etcd':
+            garbage_collect_interval = int(self._config.get("garbage_collect_interval", 10))
+            self._garbage_collector = GarbageCollect(identifier=HOST, registry=registry, interval=garbage_collect_interval)
 
-        self._v1_0_api = v1_0.Routes(logger=logger, registry=registry)
+        self._v1_0_api = v1_0.Routes(logger=logger, registry=registry, registry_type=config['registry']['type'])
```

Comment:
Could the type of registry be made a property of the registry object so we don't need to pass anything extra in here?